# Tarea de clase # 1
**Programación I**  
**Helen Saraí Avila Cabrera**

## Ejercicio 1

* Asigne valores para ``A`` y ``B``. 
* Para cada operador (``+``,``-``,``*``,``/``,``/``/,``%``,``**``) haga un ejemplo con los valores asignados previamente. 
* Utilizando el comando ``print()`` Muestre los valores a operar, la operación y el resultado.  


Definiendo $$A=10$$  y $$B=4$$

In [10]:
A=10
print(A)

10


In [11]:
B=4
print(B)

4


# Suma (+)

In [4]:
print("A+B = ", A+B)

A+B =  14


# Resta (-)

In [6]:
print("A-B = ",A-B)

A-B =  6


# Multiplicación (*)

In [7]:
print("A*B = ", A*B)

A*B =  40


# División Float (/)

In [8]:
print("A/B = ", A/B)

A/B =  2.5


# División Entera (//)

In [9]:
print("A//B = ", A//B)

A//B =  2


# Residuo (%)

In [12]:
print("A%B = ", A%B)

A%B =  2


# Potencia (**)

In [13]:
print("A**2 = ", A**2)

A**2 =  100


In [14]:
print("B**3 = ", B**3)

B**3 =  64


### Subejercicio

Cómo usar el operador de resto para determinar si un número es múltiplo de otro. 

In [17]:
x=12
y=3
z=5
print("x =", x)
print ("y =", y)
print ("z =", z)

x = 12
y = 3
z = 5


In [19]:
prueba1 = x%y
prueba2 = x%z
print("prueba1 =", prueba1)
print("prueba2 =", prueba2)

prueba1 = 0
prueba2 = 2


*Para comprobar si un valor es entero o no, se compararan los resultados con el valor "0", en donde si el resultado obtenido es "True" quiere decir que tenemos un valor entero*

In [20]:
print("prueba1 =", prueba1==0)
print("prueba2 =", prueba2==0)

prueba1 = True
prueba2 = False


## Ejercicio 2

* Asigne ``A='Aa'`` y ``B='aa'``
* Verifique las siguientes expresiones:
    - ``A<A``
    - ``A<B``
* Explique el resultado de las expresiones anteriores. ¿Por qué no son iguales?

*Definiendo A y B*

In [23]:
A = "Aa"
B = "aa"
print("A =", A)
print("B =", B)

A = Aa
B = aa


*Verificando expresiones*

In [24]:
print("A<A =", A<A)
print("A<B =", A<B)

A<A = False
A<B = True


*Explicación de resultados*

Los resultados obtenidos provienen de la aplicación del código ASCII en donde A = 65 y a = 97

In [28]:
print(ord("A"))
print(ord("a"))

65
97


In [25]:
print("A =", 65*97)
print("B =", 97*97)

A = 6305
B = 9409


Por lo tanto 6,305 **no es menor** a 6,305 y 6,305 **si es menor** a 9,409

## Ejercicio 3

1.	Escriba un programa que calcule la hipotenusa e imprima el resultado dado dos catetos. 
2.	Reescriba el ejemplo anterior para que los puntos sean bidimensionales. 
3.	Escriba un programa que, dados tres puntos bidimensionales, imprima si son colineales o no. (No emplee la estructura ``if else``, solamente operadores).


Determinación de los catetos

In [29]:
catetoA=5
catetoB=6
print("Cateto A =", catetoA)
print("Cateto B =", catetoB)

Cateto A = 5
Cateto B = 6


Determinación de la hipotenusa

In [31]:
Hipotenusa = ((catetoA)**2+(catetoB)**2)**0.5
print("Hipotenusa = ", Hipotenusa)

Hipotenusa =  7.810249675906654


# Hipotenusa con puntos bidimensionales

El punto C estará definido como (1,1) y el punto D estará definido como (3,4), por lo tanto tenemos

In [39]:
X1=1
Y1=1
X2=3
Y2=4
#definiendo los catetos tenemos:
CatetoC=X2-X1
CatetoD=Y2-Y1
print("Cateto C =", CatetoC)
print("Cateto D =", CatetoD)
Hipotenusa = ((CatetoC)**2+(CatetoD)**2)**0.5
print("Hipotenusa = ", Hipotenusa)
print("La hipotenusa es", Hipotenusa)

Cateto C = 2
Cateto D = 3
Hipotenusa =  3.605551275463989
La hipotenusa es 3.605551275463989


### Escriba un programa que, dados tres puntos bidimensionales, imprima si son colineales o no

In [46]:
f = (1,1)
g = (5,2)
h = (3,4)

i = (g[0]-f[0], g[1]-f[1])
j = (g[0]-h[0], g[1]-h[1])

print("Distancia 1 =",i)
print("Distancia 2 =",j)

Ppunto = (i[0]*j[0]+i[1]*j[1])

print("El producto punto entre la distancia 1 y 2 es igual a ", Ppunto)

Normai = (i[0]**2 + i[1]**2)**0.5
Normaj = (j[0]**2 + j[1]**2)**0.5

print("La norma de la distancia 1 es ", Normai)
print("La norma de la distancia 2 es ", Normaj)

Colin = Ppunto/(Normai*Normaj)
print("La colinealidad encontrada es ", Colin)

#El programa determinara si es colineal si el resultado de comparar "Colin" con el valor "1" es "True"

print("Los puntos son colineales: ", Colin ==1)

Distancia 1 = (4, 1)
Distancia 2 = (2, -2)
El producto punto entre la distancia 1 y 2 es igual a  6
La norma de la distancia 1 es  4.123105625617661
La norma de la distancia 2 es  2.8284271247461903
La colinealidad encontrada es  0.5144957554275265
Los puntos son colineales:  False


## Ejercicio 4

Traduzca las siguientes expresiones lógicas y verifique su resultado, siendo ``A=0`` y ``B=1``. 

1. $A \implies B$
2. $A \iff B$

Como A = 0 y B = 1  
El equivalente lógicos de la implicación es igual a $$\neg A \vee B$$  y la doble implicación es equivalente a  $$(A \implies B) \wedge (B \implies A)$$  
Ahora bien la negación , and,  or son:  

In [59]:
A=0
B=1
negA = 1 - A
negB = 1 - B
AyB = A * B
AoB = A + B - (A * B)

print("Negación A = ",negA)
print("A y B = ", AyB)
print("A o B = ", AoB)

print("El resultado de 𝐴⟹𝐵 es ", negA + B - (negA * B))
print("El resultado de 𝐴⟺𝐵 es ", (negA + B - (negA * B))*(negB + A - (negB*A)))

Negación A =  1
A y B =  0
A o B =  1
El resultado de 𝐴⟹𝐵 es  1
El resultado de 𝐴⟺𝐵 es  0
